File: python/191-200/House Robber.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def rob(self, nums):
        if not nums:
            return 0
        if(len(nums) == 1):
            return nums[0]
        
        p = [0]*len(nums)
        p[0] = nums[0]
        p[1] = max(nums[0], nums[1])
        
        for i in range(2, len(nums)):
            p[i] = max(p[i-2] + nums[i], p[i-1])
        
        return p[-1]

File: cpp/191-200/House Robber.cpp
class Solution {
public:
    int rob(vector<int> &num) {
        if (num.empty()) return 0;
        
        int p[num.size()];
        p[0] = num[0];
        p[1] = max(num[0], num[1]);
        
        for (int i = 2;i < num.size();i++)
            p[i] = max(p[i-2] + num[i], p[i-1]);
        return p[num.size()-1];
    }
};
